src/main/java/com/mraof/minestuck/blockentity/TransportalizerBlockEntity.java

```diff
@@ -212,21 +215,58 @@ private static boolean isDimensionForbidden(Level level)
 				|| typeKey.isPresent() && forbiddenDimTypes.contains(String.valueOf(typeKey.get().location()));
 	}
 	
+	public boolean hasId()
+	{
+		return !id.isEmpty();
+	}
+	
 	public String getId()
 	{
 		return id;
 	}
 	
+	/**
+	 * Attempt to set the id, failing if the id has already been taken
+	 * Does nothing if the transportalizer is locked or already has an id
+	 */
+	public void trySetId(String id, Player player)
+	{
+		if(this.active && this.hasId() || this.locked)
+			return;
+		
+		if(level != null && !level.isClientSide && !id.isEmpty())
+		{
+			var data = TransportalizerSavedData.get(level);
+			if(data.get(id) != null)
+			{
+				player.sendSystemMessage(Component.translatable(TAKEN, id));
+				return;
+			}
+			
+			this.id = id;
+			GlobalPos location = GlobalPos.of(level.dimension(), worldPosition);
+			BlockState state = level.getBlockState(worldPosition);
+			this.setChanged();
+			level.sendBlockUpdated(worldPosition, state, state, 0);
+			data.set(id, location);
+		}
+	}
+	
+	/**
+	 * Set the id, deactivating the transportalizer if it is already taken
+	 */
 	public void setId(String id)
 	{
-		if(level != null && !level.isClientSide)
+		if(level != null && !level.isClientSide && !id.isEmpty())
 		{
+			var data = TransportalizerSavedData.get(level);
 			GlobalPos location = GlobalPos.of(level.dimension(), worldPosition);
-			if(active && !this.id.isEmpty())
-				TransportalizerSavedData.get(level).remove(this.id, location);
```

Comment:
Removing this code is kinda bad. I suggest that you either:
- Add it back, so that `setId()` can replace an already-set id without causing issues.
- Add a `this.hasId()` check so that `setId()` will only set the id if an id has not already been set.